File: core/src/main/java/com/mynextduty/core/service/AuthService.java
package com.mynextduty.core.service;

import com.mynextduty.core.dto.GlobalMessageDto;
import com.mynextduty.core.dto.auth.AuthRequestDto;
import com.mynextduty.core.dto.auth.AuthResponseDto;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;

public interface AuthService {
  String publicKey();

  AuthResponseDto login(AuthRequestDto authRequestDto, HttpServletResponse httpServletResponse);

  AuthResponseDto refreshToken(HttpServletRequest request, HttpServletResponse httpServletResponse);

  GlobalMessageDto logout(HttpServletRequest httpServletRequest);
}

File: lens/src/service/auth.service.js
import api from "../service/axiosInstance";
import { API_URLS } from "../service/apiUrls";

export const authService = {
  login(payload) {
    return api.post("/auth/login", payload);
  },

  signup(payload) {
    return api.post("/auth/signup", payload);
  },

  logout() {
    return api.post("/auth/logout");
  },
};
